<!-- -*- mode: XML -*- -->
<launch>

<arg name="time_now" default="temp" />
<arg name="gnssPortPath" default="/dev/gnss"/>
<arg name="gpsdIp" default="localhost"/>
<arg name="gpsdPort" default="2947"/>

<arg name="loggerPath" default="/opt/Poseidon/www/webroot/record/"/>
<arg name="configPath" default="/opt/Poseidon/config.txt"/>

<arg name="boardVersion" value="2.2"/>

<node pkg="i2c_controller" name="i2c_controller" type="i2c_controller_node" output="screen" args="$(arg boardVersion)" respawn="true" respawn_delay="1"/>

<node pkg="hydroball_config_websocket" name="hydroball_config_websocket" type="hydroball_config_websocket_node" args="$(arg configPath)" output="screen" respawn="true" respawn_delay="1"/>

<node ns="imu" name="imu_node" pkg="imu_bno055" type="bno055_i2c_node" respawn="true" respawn_delay="2" output="screen">
        <param name="device" type="string" value="/dev/i2c-1"/>
        <param name="address" type="int" value="40"/> <!-- 0x28 == 40 is the default for BNO055 -->
        <param name="frame_id" type="string" value="imu"/>
	<param name="calibrationFile" type="string" value="/opt/Poseidon/calibration.dat" />
</node>

<node pkg="gpsd_client" name="GPS" type="gpsd_client" output="screen" respawn="true" respawn_delay="1" args="$(arg gpsdIp) $(arg gpsdPort)"/>

<node pkg="gnss_zed_f9p" name="gnss_zed_f9p" type="gnss_zed_f9p_node" output="screen" respawn="true" respawn_delay="1" args="$(arg loggerPath) $(arg gnssPortPath)"/>


<node pkg="sonar_nmea_0183_tcp_client" type="nmea_device_node" name="Sonar_NMEA" output="screen" respawn="true" respawn_delay="1">
	<param name="ip_address" value="127.0.0.1"/>
	<param name="port" value="5000"/>
</node>
 
<node pkg="wifi_file_transfer_config" type="wifi_config_node.py" name="wifi_config" output="screen" respawn="true" respawn_delay="1"/>

<node pkg="raspberrypi_vitals" name="Vitals" type="raspberrypi_vitals_node" output="screen" respawn="true" respawn_delay="1"/>

<node pkg="state_controller" name="state_controller" type="state_controller_node" output="screen" respawn="true" respawn_delay="1"/>

<node pkg="logger" name="logger" type="logger_text_node" output="screen" args="$(arg loggerPath)" respawn="true" respawn_delay="1">
	<param name="fileExtensionForGpsDatagram" value=".ubx"/>
</node>

<node pkg="hydroball_data_websocket" name="hydroball_data_websocket" type="hydroball_data_websocket_node" args="$(arg loggerPath)" respawn="true" respawn_delay="1"/>

<node pkg="hydroball_files_websocket" name="hydroball_files_websocket" type="hydroball_files_websocket_node" args="$(arg loggerPath)" respawn="true" respawn_delay="1"/>

<node pkg="diagnostics" name="Diagnostics" type="diagnostics_websocket.py" output="screen" respawn_delay="1"/>

</launch>
